devlib/target.py

```diff
@@ -582,6 +592,22 @@ def _install_module(self, mod, **params):
     def _resolve_paths(self):
         raise NotImplementedError()
 
+    # Some connections (i.e. gem5) have an internal, more efficient method of
+    # capturing the screen so try that first
+    def _overwritten_capture_screen(self, filepath):
```

Comment:
This should be implemented in the platform  and monkey patched in to the target.